nixopsaws/backends/ec2.py

```diff
@@ -1790,10 +1790,8 @@ def testzone(hosted_zone, zone):
             raise Exception("hosted zone for {0} not found".format(hosted_zone))
 
         # use hosted zone with longest match
-        zones = sorted(
-            zones, cmp=lambda a, b: cmp(len(a.Name), len(b.Name)), reverse=True
-        )
-        zoneid = zones[0]["Id"].split("/")[2]
+        longest_zone = max(zones, key=lambda x: len(x.Name))
+        zoneid = longest_zone["Id"].split("/")[2]
```

Comment:
```
>>> a = { "Name": "Foo", "Val": 2 }
>>> b = { "Name": "FooBar", "Val": 1 }

>>> max([a,b], key=lambda x: len(x['Name']))
{'Name': 'FooBar', 'Val': 1}

>>> max([a,b], key=lambda x: x['Val'])
{'Name': 'Foo', 'Val': 2}
```
Combined with `2to3`, I was able to provision my AWS cluster with NixOps on the master branch and this commit.